From d4e70f1076dd60597b1a2e4da6eb9ca6158ef595 Mon Sep 17 00:00:00 2001
From: Holotronic <producer@holotronic.dk>
Date: Wed, 21 Jul 2021 22:00:07 +0200
Subject: [PATCH 04/10] add .dtb files to kernel-image .deb

Signed-off-by: Holotronic <producer@holotronic.dk>
---
 scripts/package/builddeb | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/scripts/package/builddeb b/scripts/package/builddeb
index c4c580f547ef..38c981a9d1de 100755
--- a/scripts/package/builddeb
+++ b/scripts/package/builddeb
@@ -80,6 +80,7 @@ BUILD_DEBUG=$(if_enabled_echo CONFIG_DEBUG_INFO Yes)
 rm -rf "$tmpdir" "$kernel_headers_dir" "$libc_headers_dir" "$dbg_dir" $objtree/debian/files
 mkdir -m 755 -p "$tmpdir/DEBIAN"
 mkdir -p "$tmpdir/lib" "$tmpdir/boot"
+mkdir -p "$tmpdir/lib" "$tmpdir/boot/dtb"
 mkdir -p "$kernel_headers_dir/lib/modules/$version/"
 
 # Build and install the kernel
@@ -94,6 +95,7 @@ else
 	cp $KCONFIG_CONFIG "$tmpdir/boot/config-$version"
 fi
 cp "$($MAKE -s -f $srctree/Makefile image_name)" "$tmpdir/$installed_image_path"
+cp arch/$ARCH/boot/dts/*.dtb "$tmpdir/boot/dtb"
 
 if is_enabled CONFIG_OF_EARLY_FLATTREE; then
 	# Only some architectures with OF support have this target
-- 
2.30.2

