From c7df1eb3e2636f06bcc946e845a0ef6b6edfcad1 Mon Sep 17 00:00:00 2001
From: Michael Brown <producer@holotronic.dk>
Date: Tue, 1 May 2018 15:37:43 +0200
Subject: [PATCH 4/5] Enable cpu DMA(0-3), fix for audio drivers

Signed-off-by: Michael Brown <producer@holotronic.dk>
---
 include/config_distro_bootcmd.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/config_distro_bootcmd.h b/include/config_distro_bootcmd.h
index 66e1065..c1ca235 100644
--- a/include/config_distro_bootcmd.h
+++ b/include/config_distro_bootcmd.h
@@ -338,7 +338,8 @@
 	\
 	"boot_extlinux="                                              \
         "if test ${fpgaload_on_boot} = 1;"                         \
-        "then run fpgaload; bridge enable; fi;"       \
+        "then run fpgaload; bridge enable;"       \
+        "mw.b 0xffd05018 0xf0; fi;"       \
 		"sysboot ${devtype} ${devnum}:${distro_bootpart} any "    \
 			"${scriptaddr} ${prefix}extlinux/extlinux.conf\0" \
 	\
-- 
2.7.4

