From 3f5a6f9f05b2ce427458a0cf8c95f72bc452ede8 Mon Sep 17 00:00:00 2001
From: Michael Brown <producer@holotronic.dk>
Date: Sat, 14 Jul 2018 11:12:23 +0200
Subject: [PATCH 3/4] add generate /boot/uEnv.txt from kernel install

Signed-off-by: Michael Brown <producer@holotronic.dk>
---
 scripts/package/builddeb | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/scripts/package/builddeb b/scripts/package/builddeb
index 0bc87473f68f..7ff5f070212f 100755
--- a/scripts/package/builddeb
+++ b/scripts/package/builddeb
@@ -193,6 +193,13 @@ if grep -q '^CONFIG_BLK_DEV_INITRD=y' $KCONFIG_CONFIG; then
 else
 	want_initrd=No
 fi
+
+# create uEnv.txt containing current kernelversio in /boot folder (for u-boot probing)
+cat <<EOF > "$tmpdir/boot/uEnv.txt"
+linuximage=/boot/vmlinuz-${version}
+fdtfile=/usr/lib/linux-image-${version}/xilinx/zynqmp-zcu100-revC.dtb
+EOF
+
 for script in postinst postrm preinst prerm ; do
 	mkdir -p "$tmpdir$debhookdir/$script.d"
 	cat <<EOF > "$tmpdir/DEBIAN/$script"
-- 
2.17.1

