From 45754ee22efab3045089120ddbcdf997212b35fa Mon Sep 17 00:00:00 2001
From: Holotronic <producer@holotronic.dk>
Date: Tue, 13 Oct 2020 16:13:03 +0200
Subject: [PATCH 03/12] add .dtb files to kernel-image .deb

Signed-off-by: Holotronic <producer@holotronic.dk>
---
 scripts/package/builddeb | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/scripts/package/builddeb b/scripts/package/builddeb
index e15159d0a884..b6d3675e8087 100755
--- a/scripts/package/builddeb
+++ b/scripts/package/builddeb
@@ -126,7 +126,7 @@ BUILD_DEBUG="$(grep -s '^CONFIG_DEBUG_INFO=y' $KCONFIG_CONFIG || true)"
 # Setup the directory structure
 rm -rf "$tmpdir" "$kernel_headers_dir" "$libc_headers_dir" "$dbg_dir" $objtree/debian/files
 mkdir -m 755 -p "$tmpdir/DEBIAN"
-mkdir -p "$tmpdir/lib" "$tmpdir/boot"
+mkdir -p "$tmpdir/lib" "$tmpdir/boot/dtb"
 mkdir -p "$kernel_headers_dir/lib/modules/$version/"
 
 # Build and install the kernel
@@ -141,6 +141,7 @@ else
 	cp $KCONFIG_CONFIG "$tmpdir/boot/config-$version"
 fi
 cp "$($MAKE -s image_name)" "$tmpdir/$installed_image_path"
+cp arch/$ARCH/boot/dts/*.dtb "$tmpdir/boot/dtb"
 
 if grep -q "^CONFIG_OF=y" $KCONFIG_CONFIG ; then
 	# Only some architectures with OF support have this target
-- 
2.25.1

