From 32fc7e353265ecba207d096aeb4c9bb9c2ffe872 Mon Sep 17 00:00:00 2001
From: Michael Brown <producer@holotronic.dk>
Date: Wed, 11 Jul 2018 18:11:35 +0200
Subject: [PATCH] Modify env variables to pick up kernel deb boot files

Signed-off-by: Michael Brown <producer@holotronic.dk>
---
 include/configs/xilinx_zynqmp.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/include/configs/xilinx_zynqmp.h b/include/configs/xilinx_zynqmp.h
index 59caeae395..33dab96b37 100644
--- a/include/configs/xilinx_zynqmp.h
+++ b/include/configs/xilinx_zynqmp.h
@@ -156,11 +156,11 @@
 	"fdt_offset=0x200000\0" \
 	"kernel_size=0x1e00000\0" \
 	"fdt_size=0x80000\0" \
-	"bootenv=uEnv.txt\0" \
+	"bootenv=/boot/uEnv.txt\0" \
 	"loadbootenv=load mmc $sdbootdev:$partid ${loadbootenv_addr} ${bootenv}\0" \
 	"importbootenv=echo Importing environment from SD ...; " \
 		"env import -t ${loadbootenv_addr} $filesize\0" \
-	"sd_uEnvtxt_existence_test=test -e mmc $sdbootdev:$partid /uEnv.txt\0" \
+	"sd_uEnvtxt_existence_test=test -e mmc $sdbootdev:$partid /boot/uEnv.txt\0" \
 	"sata_root=if test $scsidevs -gt 0; then setenv bootargs $bootargs root=/dev/sda rw rootfstype=ext4; fi\0" \
 	"sataboot=load scsi 0 80000 boot/Image && load scsi 0 $fdt_addr boot/system.dtb && booti 80000 - $fdt_addr\0" \
 	"veloce=fdt addr f000000 && fdt resize" \
@@ -183,8 +183,8 @@
 			"run uenvcmd; " \
 		"fi\0" \
 	"sdboot=mmc dev $sdbootdev && mmcinfo && run uenvboot || run sdroot$sdbootdev; " \
-		"load mmc $sdbootdev:$partid $fdt_addr system.dtb && " \
-		"load mmc $sdbootdev:$partid $kernel_addr Image && " \
+		"load mmc $sdbootdev:$partid $fdt_addr $fdtfile && " \
+		"load mmc $sdbootdev:$partid $kernel_addr $linuximage && " \
 		"booti $kernel_addr - $fdt_addr\0" \
 	"emmcboot=run sdboot\0" \
 	"nandboot=nand info && nand read $fdt_addr $fdt_offset $fdt_size && " \
-- 
2.17.1

