From 09479d9d7c059b4801e46115c6281e9a4d17259c Mon Sep 17 00:00:00 2001
From: Michael Brown <producer@holotronic.dk>
Date: Sun, 8 Apr 2018 17:28:06 +0200
Subject: [PATCH 2/2] Add support for fw_printenv fw_setenv

Signed-off-by: Michael Brown <producer@holotronic.dk>
---
 configs/socfpga_de1_soc_defconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configs/socfpga_de1_soc_defconfig b/configs/socfpga_de1_soc_defconfig
index bc7af51..19e3b7b 100644
--- a/configs/socfpga_de1_soc_defconfig
+++ b/configs/socfpga_de1_soc_defconfig
@@ -57,3 +57,4 @@ CONFIG_USB_DWC2=y
 CONFIG_USB_STORAGE=y
 CONFIG_USE_TINY_PRINTF=y
 # CONFIG_EFI_LOADER is not set
+CONFIG_FILE=y
-- 
2.7.4

