From fc414cbdbc38122c0f637c7c5caf2aafe52dff67 Mon Sep 17 00:00:00 2001
From: Michael Brown <producer@holotronic.dk>
Date: Mon, 2 Apr 2018 23:59:02 +0200
Subject: [PATCH 3/4] add generate /boot/extlinux/extlinux.conf for uboot

Signed-off-by: Michael Brown <producer@holotronic.dk>
---
 scripts/package/builddeb | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/scripts/package/builddeb b/scripts/package/builddeb
index b4f0f2b..1c2f095 100755
--- a/scripts/package/builddeb
+++ b/scripts/package/builddeb
@@ -193,6 +193,16 @@ if grep -q '^CONFIG_BLK_DEV_INITRD=y' $KCONFIG_CONFIG; then
 else
 	want_initrd=No
 fi
+
+# create extlinux.conf  (for u-boot probing)
+mkdir -p "$tmpdir/boot/extlinux"
+cat <<EOF > "$tmpdir/boot/extlinux/extlinux.conf"
+label Linux ${version}
+    kernel /boot/vmlinuz-$version
+    append  console=ttyS0,115200 root=/dev/mmcblk0p3 rw rootfstype=ext4 rootwait
+    fdtdir /usr/lib/$packagename
+EOF
+
 for script in postinst postrm preinst prerm ; do
 	mkdir -p "$tmpdir$debhookdir/$script.d"
 	cat <<EOF > "$tmpdir/DEBIAN/$script"
-- 
2.7.4

