From c16e72696608e2f4e84ed00c5359688013880302 Mon Sep 17 00:00:00 2001
From: Holotronic <producer@holotronic.dk>
Date: Wed, 21 Jul 2021 22:08:12 +0200
Subject: [PATCH 06/10] Enable bridges in dtb's

Signed-off-by: Holotronic <producer@holotronic.dk>
---
 arch/arm/boot/dts/socfpga.dtsi | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/boot/dts/socfpga.dtsi b/arch/arm/boot/dts/socfpga.dtsi
index e40422077438..2e3591881422 100644
--- a/arch/arm/boot/dts/socfpga.dtsi
+++ b/arch/arm/boot/dts/socfpga.dtsi
@@ -531,6 +531,7 @@
 			reg = <0xff400000 0x100000>;
 			resets = <&rst LWHPS2FPGA_RESET>;
 			clocks = <&l4_main_clk>;
+			bridge-enable = <1>;
 		};
 
 		fpga_bridge1: fpga_bridge@ff500000 {
@@ -538,6 +539,7 @@
 			reg = <0xff500000 0x10000>;
 			resets = <&rst HPS2FPGA_RESET>;
 			clocks = <&l4_main_clk>;
+			bridge-enable = <1>;
 		};
 
 		fpgamgr0: fpgamgr@ff706000 {
-- 
2.30.2

